language-support/hs/bindings/src/DA/Ledger/Types.hs

```diff
@@ -210,30 +217,32 @@ data RecordField
 
 data Variant
     = Variant {
-        vid         :: VariantId,
-        constructor :: ConstructorId,
-        value       :: Value } deriving (Eq,Ord,Show)
+        vid   :: Maybe Identifier,
+        cons  :: ConstructorId,
+        value :: Value } deriving (Eq,Ord,Show)
 
 data Identifier
     = Identifier {
         pid :: PackageId,
         mod :: ModuleName,
         ent :: EntityName } deriving (Eq,Ord,Show)
 
-newtype MicroSecondsSinceEpoch = MicroSecondsSinceEpoch Int deriving (Eq,Ord,Show)-- TODO: Int64?
```

Comment:
Using `Int64` is a good idea. Most of the numbers we get here would not fit into an `Int32`.

Reply:
Will `Int` ever realistically be `Int32` ?

Reply:
The only guarantee the Haskell 2010 report gives is that `Int` has at least 30 bits. Since using `Int64` doesn't hurt, I'd prefer to do it.

Reply:
ok